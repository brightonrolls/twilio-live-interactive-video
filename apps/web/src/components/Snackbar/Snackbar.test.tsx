import { shallow } from 'enzyme';
import React from 'react';
import { SnackbarImpl } from './Snackbar';

describe('the Snackbar component', () => {
  it('should render correctly with "warning" variant', () => {
    const wrapper = shallow(
      <SnackbarImpl variant="warning" headline="Test Headline" message="Test Message" handleClose={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with "error" variant', () => {
    const wrapper = shallow(
      <SnackbarImpl variant="error" headline="Test Headline" message="Test Message" handleClose={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with "info" variant', () => {
    const wrapper = shallow(
      <SnackbarImpl variant="info" headline="Test Headline" message="Test Message" handleClose={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with no handleClose function provided', () => {
    const wrapper = shallow(<SnackbarImpl variant="error" headline="Test Headline" message="Test Message" />);
    expect(wrapper).toMatchSnapshot();
  });
});
