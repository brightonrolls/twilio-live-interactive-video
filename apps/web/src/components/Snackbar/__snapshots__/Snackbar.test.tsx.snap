// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Snackbar component should render correctly with "error" variant 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  autoHideDuration={10000}
  onClose={[Function]}
  open={true}
>
  <SnackbarImpl
    handleClose={[Function]}
    headline="Test Headline"
    message="Test Message"
    variant="error"
  />
</WithStyles(ForwardRef(Snackbar))>
`;

exports[`the Snackbar component should render correctly with "info" variant 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  autoHideDuration={10000}
  onClose={[Function]}
  open={true}
>
  <SnackbarImpl
    handleClose={[Function]}
    headline="Test Headline"
    message="Test Message"
    variant="info"
  />
</WithStyles(ForwardRef(Snackbar))>
`;

exports[`the Snackbar component should render correctly with "warning" variant 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  autoHideDuration={10000}
  onClose={[Function]}
  open={true}
>
  <SnackbarImpl
    handleClose={[Function]}
    headline="Test Headline"
    message="Test Message"
    variant="warning"
  />
</WithStyles(ForwardRef(Snackbar))>
`;

exports[`the Snackbar component should render correctly with no handleClose function provided 1`] = `
<WithStyles(ForwardRef(Snackbar))
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "top",
    }
  }
  autoHideDuration={10000}
  onClose={[Function]}
  open={true}
>
  <SnackbarImpl
    headline="Test Headline"
    message="Test Message"
    variant="error"
  />
</WithStyles(ForwardRef(Snackbar))>
`;
